### ISSUE TITLE
(custom-resources): No way to rename log group/lambda for AwsCustomResource

### ISSUE THREAD

**Issue by ckifer** [aws/aws-cdk#19309]
### General Issue

No way to rename log group/lambda for AwsCustomResource

### The Question

I'm not sure if this is a feature or bug, sorry if this is in the wrong place.

`AwsCustomResource` uses a SingletonLambda to make a single SDK call (I didn't, understand this when I started my project). I utilize this a few times in the same stack for a website:
1. Get my AppMonitor ID for CW RUM (not in CFN spec yet)
2. Invalidate my CloudFront distribution at the end of deploy
3. Upload some JSON to my website bucket as config for the site

I created one of these first, say a custom resource of the following form:
```typescript
  private getAppMonitorId() {
    const awsRUMSDKCall: AwsSdkCall = {
      service: 'RUM',
      action: 'getAppMonitor',
      parameters: { Name: this.rumMonitor.name },
      physicalResourceId: PhysicalResourceId.of('7854f358-6c0a-4ed4-8863-2d5bf35d7062'),
    };

    const customResource = new AwsCustomResource(this, 'Custom::GetAppMonitorId', {
      policy: AwsCustomResourcePolicy.fromSdkCalls({
        resources: [this.rumMonitorArn],
      }),
      installLatestAwsSdk: true,
      onCreate: awsRUMSDKCall,
      onUpdate: awsRUMSDKCall,
      functionName: 'GetAppMonitorCustomResourceHandler',
    });

    customResource.node.addDependency(this.rumMonitor);

    return customResource.getResponseField('AppMonitor.Id');
  }
}
```
The `functionName` here names the Singleton Lambda and the log group that gets created `/aws/lambda/GetAppMonitorCustomResourceHandler`. This is all well and good until another custom resource is added, with a `functionName` or not. The new custom resource then uses the named lambda and the named log group named specifically for a different custom resource. This creates confusion when trying to debug, trying to look for the logs (especially if you don't know that `AwsCustomResource` is a singleton lambda), etc.

Out of this, I ask two things:
1. Is it possible to rename an `AwsCustomResource`? 
   * I think the answer to this is no, as I get `Modifying service token is not allowed.` on any attempts I make.
2. I would move to suggest to remove/deprecate `functionName` from `AwsCustomResource` as the creation of a specifically named lambda that potentially adds unrelated `AwsCustomResource`s to it is confusing. I don't know what implications that removal would have and is probably a breaking change so I'm not too worried about it now that I know. Calling it out at the top of the `AwsCustomResource` docs page would be nice at least.

### CDK CLI Version

2.0.0

### Framework Version

1.147.0

### Node.js Version

16

### OS

Mac OSX Monterey

### Language

Typescript

### Language Version

4.0.5

**Comment by peterwoodworth** [comment]
Hey @ckifer,

You're right, as you've found, the AwsCustomResource will create a Singleton function to ensure that only one custom resource handler will be created no matter how many times you use the AwsCustomResource construct. This means your function will stick to the same name as the first AwsCustomResource declaration you made. Here's the declaration of the SingletonFunction in our code:

https://github.com/aws/aws-cdk/blob/f9b35cf9975f741d4baebf0ae210da414c779a13/packages/%40aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.ts#L360-L370

According to [singleton function docs,](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-lambda.SingletonFunction.html) a singleton function will only be added once, assuming that the `uuid` and `lambdaPurpose` props stay the same. Which in this case they are guaranteed to stay the same, since they are hardcoded as seen above. 

The logical ID is also hardcoded, so there's no way ([bar escape hatches](https://bobbyhadz.com/blog/override-logical-id-aws-cdk)) to work around the limitations of using physical resource names imposed by [the way CloudFormation works](https://docs.aws.amazon.com/cdk/v2/guide/resources.html#resources_physical_names)

So in short, Regarding 1) you can rename this function through a couple means, none of which are ideal. One way is to use escape hatches to override the logical ID of the singleton function to trigger the creation of just a new Lambda function with a new name. Another way is to delete your stack, rename the function with the `functionName` prop, then redeploy.
And, the problems you see here with question 2) are going to be there regardless of if we let the user use the functionName prop or not.

And yes, we need to improve our docs on this, thanks for calling that out; we barely call out that we use a singleton function here at all. I'll try to get to that in the next couple weeks

**Comment by ckifer** [comment]
Thanks for the detailed response and explanation @peterwoodworth! Makes a lot of sense after my woes of trying to figure where my logs were going. 

As long as the docs are improved on this I am good with the resolution 👍🏼
